onnxruntime/core/providers/webgpu/tensor/slice.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/webgpu/webgpu_kernel.h"
+#include "core/providers/webgpu/program.h"
+#include <iostream>
+
+namespace onnxruntime {
+namespace webgpu {
+
+class SliceProgram final : public Program<SliceProgram> {
+ public:
+  SliceProgram() : Program{"Slice"} {}
+
+  Status GenerateShaderCode(ShaderHelper& sh) const override;
+
+  WEBGPU_PROGRAM_DEFINE_UNIFORM_VARIABLES({"output_size", ProgramUniformVariableDataType::Uint32},
+                                          {"starts", ProgramUniformVariableDataType::Uint32},
+                                          {"steps", ProgramUniformVariableDataType::Uint32},
+                                          {"signs", ProgramUniformVariableDataType::Int32});
+};
+
+class Slice final : public WebGpuKernel {
+ public:
+  Slice(const OpKernelInfo& info) : WebGpuKernel(info) {
+    hasStartsAttr = info.GetAttrs("starts", attr_starts_).IsOK();
```

Comment:
is there case that the attribute contains "starts" but it's an empty list?

if not, we don't need the 3 boolean value. just use attr_starts_.empty() is good